File: VB/MyGridCtrl/MyGridInfoRegistrator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.BandedGrid

Namespace DevExpress.S170736
	Public Class MyGridInfoRegistrator
		Inherits BandedGridInfoRegistrator
		Public Overrides Function CreateView(ByVal grid As GridControl) As XtraGrid.Views.Base.BaseView
			Return New MyBandedGridView(TryCast(grid, GridControl))
		End Function
		Public Overrides Function CreateViewInfo(ByVal view As XtraGrid.Views.Base.BaseView) As XtraGrid.Views.Base.ViewInfo.BaseViewInfo
			Return New MyBandedGridViewInfo(CType(view, BandedGridView))
		End Function
		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyBandedGridView"
			End Get
		End Property
		Public Overrides Function CreateHandler(ByVal view As XtraGrid.Views.Base.BaseView) As XtraGrid.Views.Base.Handler.BaseViewHandler
			Return New MyGridHandler(TryCast(view, MyBandedGridView))
		End Function
	End Class
End Namespace

File: CS/MyGridCtrl/MyGridInfoRegistrator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.BandedGrid;

namespace DevExpress.S170736 {
    public class MyGridInfoRegistrator : BandedGridInfoRegistrator {        
        public override XtraGrid.Views.Base.BaseView CreateView(GridControl grid) {
            return new MyBandedGridView(grid as GridControl);
        }
        public override XtraGrid.Views.Base.ViewInfo.BaseViewInfo CreateViewInfo(XtraGrid.Views.Base.BaseView view) {
            return new MyBandedGridViewInfo((BandedGridView)view);
        }
        public override string ViewName {
            get {
                return "MyBandedGridView";
            }
        }
        public override XtraGrid.Views.Base.Handler.BaseViewHandler CreateHandler(XtraGrid.Views.Base.BaseView view) {
            return new MyGridHandler(view as MyBandedGridView);
        }
    }
}
